unstructured/ingest/connector/jira.py

```diff
@@ -0,0 +1,491 @@
+                    raise KeyError(
+                        "Response object has no known keys to \
+                                   access the results, such as 'results' or 'values'.",
+                    )
+                all_results += func(*args, **kwargs)[results_key]
+            kwargs["start"] += kwargs["limit"]
+
+        return all_results[:number_of_items_to_fetch]
+
+    return wrapper
+
+
+@dataclass
+class JiraIngestDoc(IngestDocSessionHandleMixin, IngestDocCleanupMixin, BaseIngestDoc):
+    """Class encapsulating fetching a doc and writing processed results (but not
+    doing the processing).
+
+    Current implementation creates a Jira connection object
+    to fetch each doc, rather than creating a it for each thread.
+    """
+
+    config: SimpleJiraConfig
+    file_meta: JiraFileMeta
+    registry_name: str = "jira"
+
+    @property
+    def source_url(self):
+        return f"{self.config.url}/browse/{self.file_meta.issue_key}"
+
+    @property
+    def record_locator(self):  # Values must be JSON-serializable
+        """A dictionary with any data necessary to uniquely identify the document on
+        the source system."""
+        return {"issue_key": self.file_meta.issue_key}
+
+    @requires_dependencies(dependencies=["atlassian"], extras="jira")
+    def get_metadata_fields(self):
+        try:
+            jira = self.session_handle.service
+            issue = jira.issue(self.file_meta.issue_key)
+            issue_fields = nested_object_to_field_getter(issue["fields"])
+
+            self.metadata_fields = {
+                "date_modified": str(issue_fields["updated"]),
+                "date_created": str(issue_fields["created"]),
+                "date_processed": str(datetime.datetime.now().time()),
+                "record_locator": self.record_locator,
+            }
+
+            self.check_exists = True
+            self.file_exists = True
+
+        except HTTPError as error:
+            if error.response.reason == "Not Found":
+                self.check_exists = True
+                self.file_exists = False
+            else:
+                logger.error(f"Error: {error} for issue {self.file_meta.issue_key}")
+
+    @property
+    def date_created(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_created"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def date_modified(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_modified"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def date_processed(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_processed"] if hasattr(self, "metadata_fields") else None
```

Comment:
rather than checking and getting metadata_fields explicitly can we make a metadata_fields [@cached_property](https://docs.python.org/3/library/functools.html#functools.cached_property). then these can just access with something like `self.metadata_fields.date_modified`

Reply:
example of how we're addressing the fsspec data properties pr: https://github.com/Unstructured-IO/unstructured/pull/1279/files#diff-439dc10065723b7989f51b27d936fc10c8437ffac6495438ce28b525696ec910